src/Models/ScaffoldGenerator.ts

```diff
@@ -177,54 +75,48 @@ export class ScaffoldGenerator {
   /**
    * Create Dockerfile & devcontainer.json
    */
-  static async generateDockerRelatedFiles(devcontainerFolderPath: string, boardFolderPath: string, boardId: string): Promise<boolean> {
+  private async generateDockerRelatedFiles(type: ScaffoldType, devcontainerFolderPath: string, templateFolderPath: string, projectTemplateType: ProjectTemplateType): Promise<boolean> {
     // Dockerfile
     const dockerfileTargetPath = path.join(
       devcontainerFolderPath, FileNames.dockerfileName);
-    if (!await sdk.FileSystem.exists(devcontainerFolderPath)) {
-      await utils.mkdirRecursively(devcontainerFolderPath);
-    }
-
-    if (await sdk.FileSystem.exists(dockerfileTargetPath)) {
-      return true;
-    }
+    if (!await FileUtility.fileExists(type, dockerfileTargetPath)) {
+      if (!await FileUtility.directoryExists(type, devcontainerFolderPath)) {
+        await FileUtility.mkdirRecursively(type, devcontainerFolderPath);
+      }
 
-    try {
-      const dockerfileSourcePath = path.join(
-        boardFolderPath, boardId, FileNames.dockerfileName);
-      const dockerfileContent = fs.readFileSync(dockerfileSourcePath, 'utf8');
-      await sdk.FileSystem.writeFile(dockerfileTargetPath, dockerfileContent);
-    } catch (error) {
-      throw new Error(`Create ${FileNames.dockerfileName} failed: ${error.message}`);
+      try {
+        const dockerfileSourcePath = path.join(
+          templateFolderPath, FileNames.dockerfileName);
+        const dockerfileContent = fs.readFileSync(dockerfileSourcePath, 'utf8');
```

Comment:
Do not use sync method

Reply:
The whole refrator will be implemented in other pr.